### DISCUSSION TITLE
Manually set ID sometimes and allow for generated at others

### DISCUSSION THREAD

**Discussion by yoshidahiro** [nhibernate/fluent-nhibernate#360] [General] [answered]
We have table that has its ID?PK generated by a trigger
Id(x => x.Cable_RL_ID).GeneratedBy.Sequence("S_CABLE_RL_ID")
the trigger is here
BEGIN
   if :NEW.cable_rl_id is NULL then
      :NEW.cable_rl_id := s_cable_rl_id.nextval;
   end if;
 END

I know if I use the following to manually set the id
 Id(x => x.Cable_RL_ID).GeneratedBy.Assigned();

However the problem arises that we are converting data over from an older system
we wanted the newer data to be generated by the sequence and the older data to be manually set

is there anyway to accommodate this?

**Comment by jlscpns** [comment, marked as answer]
Rather than solving this on Fluent NH side, an overload of ISession.Save is the way to go.
